.yarn/patches/@metamask-providers-npm-18.2.0-3616fe3170.patch

```diff
@@ -0,0 +1,38 @@
+diff --git a/dist/BaseProvider.cjs b/dist/BaseProvider.cjs
+index bcb7967af0b6369cc843a159e1c369e5d8a91f12..3ec25597d3ffd48565b6b80d0c79d0d2dd6dcfc1 100644
+--- a/dist/BaseProvider.cjs
++++ b/dist/BaseProvider.cjs
+@@ -20,6 +20,7 @@ const json_rpc_engine_1 = require("@metamask/json-rpc-engine");
+ const rpc_errors_1 = require("@metamask/rpc-errors");
+ const safe_event_emitter_1 = __importDefault(require("@metamask/safe-event-emitter"));
+ const fast_deep_equal_1 = __importDefault(require("fast-deep-equal"));
++const uuid_1 = require("uuid");
+ const messages_1 = __importDefault(require("./messages.cjs"));
+ const utils_1 = require("./utils.cjs");
+ /**
+@@ -134,9 +135,11 @@ class BaseProvider extends safe_event_emitter_1.default {
+         }
+         const payload = params === undefined || params === null
+             ? {
++                id: (0, uuid_1.v4)(),
```

Comment:
An id field was necessary to ensure that requests originating from StreamProvider could be correctly matched with their responses and therefore resolve properly.
PR to the change on the providers repo https://github.com/MetaMask/providers/pull/399

Reply:
This patch is not needed anymore look up this change https://github.com/MetaMask/metamask-extension/pull/29650/commits/5ba1eeb9c74ae2d8d1fbde706bcfaf3d6c89ae03